### PR TITLE
[SQL] add support for backup storage redundancy param to create/restore/update db commands

### PR DESCRIPTION
**Pull request by xaliciayang**
**Description**  

Enabling customer to set Backup Storage Redundancy (StorageAccountType in REST API) via --backup-storage-redundancy parameter by adding parameter to the following cmds: 
az sql db create 
az sql db update 
az sql db restore
az sql db copy 
az sql db replica create 
az sql db ltr-backup restore

**Testing Guide**  

Test all of the above commands with additional --backup-storage-redundancy parameters

**History Notes**  

[SQL] BREAKING CHANGE: az sql db create: Add --backup-storage-redundancy parameter; add warning for unspecified bsr/bsr == Geo.

Same for remaining cmds.   

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

File: src/azure-cli/azure/cli/command_modules/sql/_format.py [hunk 1]
```diff
@@ -151,10 +151,20 @@ def db_transform(result):
...
 
     return result
 
 
+def _get_external_backup_storage_redundancy(self):
+    return {
+        'LRS': 'Local',
+        'GRS': 'Geo',
+        'ZRS': 'Zone'
+    }.get(self, 'Invalid')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_format.py:165]: could you add support like https://github.com/Azure/azure-cli/pull/15208/files

File: src/azure-cli/azure/cli/command_modules/sql/_help.py [hunk 2]
```diff
@@ -300,6 +306,10 @@
     text: |
         az sql db restore --dest-name MyDest --edition GeneralPurpose --name MyAzureSQLDatabase --resource-group MyResourceGroup --server myserver --subscription MySubscription --time "2018-05-20T05:34:22"
     crafted: true
+  - name: Create a new database with geo-redundant backup storage by restoring from a backup. (autogenerated)
+    text: |
+        az sql db restore --dest-name MyDest --edition GeneralPurpose --name MyAzureSQLDatabase --resource-group MyResourceGroup --server myserver --subscription MySubscription --time "2018-05-20T05:34:22" --backup-storage-redundancy Geo
+    crafted: true
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_help.py:312]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/sql/_params.py [hunk 3]
```diff
@@ -635,6 +658,11 @@ def load_arguments(self, _):
 
         c.argument('max_size_bytes', help='The new maximum size of the database expressed in bytes.')
 
+        c.argument('storage_account_type',
+                   arg_type=backup_storage_redundancy_param_type,
+                   options_list=['--backup-storage-redundancy', '--bsr'],
+                   help='Backup storage redundancy used to store backups')
+
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:665]: avoid duplicate. use CLIArgumentType to abstract common argument type and reuse it for command

File: src/azure-cli/azure/cli/command_modules/sql/_params.py [hunk 4]
```diff
@@ -554,6 +567,11 @@ def load_arguments(self, _):
                    help='Name of the server to create the copy in.'
                    ' If unspecified, defaults to the origin server.')
 
+        c.argument('storage_account_type',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:570]: allowed value is in ['LRS', 'ZRS‘, 'GRS']? What about RA-GRS, RA-GZRS, GZRS?

Comment (xaliciayang) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:570]: Backup storage uses LRS/ZRS/GRS

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**